File: DevExpress-Demo/DevExpress.SalesDemo.Wpf/VB/DevExpress.SalesDemo.Wpf/ViewModel/ProductsViewModel.vb
Imports DevExpress.Mvvm.POCO

Namespace DevExpress.SalesDemo.Wpf.ViewModel

    Public Class ProductsViewModel
        Inherits PageViewModel

        Public Shared Function Create() As ProductsViewModel
            Return ViewModelSource.Create(Function() New ProductsViewModel())
        End Function

        Protected Sub New()
        End Sub

        Public Overridable Property DailySalesByProductViewModel As PerformanceBarChartViewModel

        Public Overridable Property UnitSalesByProductViewModel As PerformanceBarChartViewModel

        Public Overridable Property PieGaugeRangeViewModel As PieBarRangeViewModel

        Protected Overrides Sub Initialize()
            DailySalesByProductViewModel = PerformanceBarChartViewModel.Create(Products, PerformanceViewMode.Daily)
            UnitSalesByProductViewModel = PerformanceBarChartViewModel.Create(Products, PerformanceViewMode.Monthly)
            PieGaugeRangeViewModel = PieBarRangeViewModel.Create(Products)
        End Sub
    End Class
End Namespace

File: DevExpress-Demo/DevExpress.SalesDemo.Wpf/CS/DevExpress.SalesDemo.Wpf/ViewModel/ProductsViewModel.cs
using DevExpress.Mvvm.POCO;

namespace DevExpress.SalesDemo.Wpf.ViewModel {
    public class ProductsViewModel : PageViewModel {
        public static ProductsViewModel Create() {
            return ViewModelSource.Create(() => new ProductsViewModel());
        }
        protected ProductsViewModel() { }

        public virtual PerformanceBarChartViewModel DailySalesByProductViewModel { get; protected set; }
        public virtual PerformanceBarChartViewModel UnitSalesByProductViewModel { get; protected set; }
        public virtual PieBarRangeViewModel PieGaugeRangeViewModel { get; protected set; }
        protected override void Initialize() {
            DailySalesByProductViewModel = PerformanceBarChartViewModel.Create(Modules.Products, PerformanceViewMode.Daily);
            UnitSalesByProductViewModel = PerformanceBarChartViewModel.Create(Modules.Products, PerformanceViewMode.Monthly);
            PieGaugeRangeViewModel = PieBarRangeViewModel.Create(Modules.Products);
        }
    }
}
